packages/devtools/test/support/cpu_profile_test_data.dart

```diff
@@ -0,0 +1,491 @@
+  name: 'F',
+  category: 'Dart',
+  url: 'url',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 0;
+final CpuStackFrame stackFrameF2 = CpuStackFrame(
+  id: 'id_6',
+  name: 'F',
+  category: 'Dart',
+  url: 'url',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 3;
+final CpuStackFrame stackFrameC2 = CpuStackFrame(
+  id: 'id_7',
+  name: 'C',
+  category: 'Dart',
+  url: 'file:///path/to/flutter/packages/flutter/lib/src/widgets/binding.dart',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 1;
+final CpuStackFrame stackFrameC3 = CpuStackFrame(
+  id: 'id_8',
+  name: 'C',
+  category: 'Dart',
+  url: 'file:///path/to/flutter/packages/flutter/lib/src/widgets/binding.dart',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 1;
+final CpuStackFrame stackFrameG = CpuStackFrame(
+  id: 'id_9',
+  name: 'G',
+  category: 'Dart',
+  url: 'file:///path/to/flutter/packages/flutter/lib/src/widgets/binding.dart',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 1;
+
+final CpuStackFrame testStackFrame = stackFrameA
+  ..addChild(stackFrameB
+    ..addChild(stackFrameC)
+    ..addChild(stackFrameD
+      ..addChild(stackFrameE..addChild(stackFrameF..addChild(stackFrameC2)))
+      ..addChild(stackFrameF2..addChild(stackFrameC3))));
+
+const String testStackFrameStringGolden = '''
+  A - children: 1 - excl: 0 - incl: 10
+    B - children: 2 - excl: 0 - incl: 10
+      C - children: 0 - excl: 2 - incl: 2
+      D - children: 2 - excl: 2 - incl: 8
+        E - children: 1 - excl: 1 - incl: 2
+          F - children: 1 - excl: 0 - incl: 1
+            C - children: 0 - excl: 1 - incl: 1
+        F - children: 1 - excl: 3 - incl: 4
+          C - children: 0 - excl: 1 - incl: 1
+''';
+
+const String bottomUpPreMergeGolden = '''
+  C - children: 1 - excl: 2 - incl: 2
+    B - children: 1 - excl: 2 - incl: 2
+      A - children: 0 - excl: 2 - incl: 2
+
+  D - children: 1 - excl: 2 - incl: 2
+    B - children: 1 - excl: 2 - incl: 2
+      A - children: 0 - excl: 2 - incl: 2
+
+  E - children: 1 - excl: 1 - incl: 1
+    D - children: 1 - excl: 1 - incl: 1
+      B - children: 1 - excl: 1 - incl: 1
+        A - children: 0 - excl: 1 - incl: 1
+
+  C - children: 1 - excl: 1 - incl: 1
+    F - children: 1 - excl: 1 - incl: 1
+      E - children: 1 - excl: 1 - incl: 1
+        D - children: 1 - excl: 1 - incl: 1
+          B - children: 1 - excl: 1 - incl: 1
+            A - children: 0 - excl: 1 - incl: 1
+
+  F - children: 1 - excl: 3 - incl: 3
+    D - children: 1 - excl: 3 - incl: 3
+      B - children: 1 - excl: 3 - incl: 3
+        A - children: 0 - excl: 3 - incl: 3
+
+  C - children: 1 - excl: 1 - incl: 1
+    F - children: 1 - excl: 1 - incl: 1
+      D - children: 1 - excl: 1 - incl: 1
+        B - children: 1 - excl: 1 - incl: 1
+          A - children: 0 - excl: 1 - incl: 1
+
```

Comment:
Consider adding a trim call to the gode generating the golden so these expectation values don't have trailing whitespace.

Reply:
added a `trimRight` call